### PR TITLE
Adding the OAuth Authenticator and Authorizor implementations.

### PR DESCRIPTION
**Pull request by sinthuja**

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/oauth/OAuthTokenValidaterStubFactory.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (milindaperera) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/oauth/OAuthTokenValidaterStubFactory.java]: year 2017

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (milindaperera) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java]: year 2017

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	 * This creates an AuthenticationInfo object that is used for authorization. This method will validate the token and
+	 * sets the required parameters to the object.
+	 *
+	 * @param token                      that needs to be validated.
+	 * @param tokenValidationServiceStub stub that is used to call the external service.
+	 * @return AuthenticationInfo This contains the information related to authenticated client.
+	 * @throws RemoteException that triggers when failing to call the external service..
+	 */
+	private AuthenticationInfo getAuthenticationInfo(String token,
+													 OAuth2TokenValidationServiceStub tokenValidationServiceStub) throws RemoteException {
```
Comment (milindaperera) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java]: Formatting

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/internal/AuthenticationServiceComponent.java [hunk 4]
```diff
@@ -80,4 +96,27 @@ protected void setRealmService(RealmService realmService) {
...
+    }
+
+    private int getPortOffset() {
+        ServerConfiguration carbonConfig = ServerConfiguration.getInstance();
+        String portOffset = System.getProperty("portOffset",
+                carbonConfig.getFirstProperty(CARBON_CONFIG_PORT_OFFSET));
+        try {
+            if ((portOffset != null)) {
+                return Integer.parseInt(portOffset.trim());
+            } else {
```
Comment (milindaperera) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/internal/AuthenticationServiceComponent.java]: can remove this else

File: components/andes/org.wso2.carbon.andes.authorization/src/main/java/org/wso2/carbon/andes/authorization/internal/AuthorizationServiceComponent.java [hunk 5]
```diff
@@ -93,4 +107,27 @@ protected void setRealmService(RealmService realmService) {
...
+    }
+
+    private int getPortOffset() {
+        ServerConfiguration carbonConfig = ServerConfiguration.getInstance();
+        String portOffset = System.getProperty("portOffset",
+                carbonConfig.getFirstProperty(CARBON_CONFIG_PORT_OFFSET));
+        try {
+            if ((portOffset != null)) {
+                return Integer.parseInt(portOffset.trim());
+            } else {
```
Comment (milindaperera) [inline components/andes/org.wso2.carbon.andes.authorization/src/main/java/org/wso2/carbon/andes/authorization/internal/AuthorizationServiceComponent.java]: can remove this else

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/oauth/OAuthTokenValidaterStubFactory.java (hunk 1 above)
Comment (sinthuja) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/oauth/OAuthTokenValidaterStubFactory.java]: Fixed.

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java (hunk 2 above)
Comment (sinthuja) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java]: Fixed.

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java (hunk 3 above)
Comment (sinthuja) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/andes/OAuth2BasedMQTTAuthenticator.java]: Fixed.

File: components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/internal/AuthenticationServiceComponent.java (hunk 4 above)
Comment (sinthuja) [inline components/andes/org.wso2.carbon.andes.authentication/src/main/java/org/wso2/carbon/andes/authentication/internal/AuthenticationServiceComponent.java]: Fixed.

File: components/andes/org.wso2.carbon.andes.authorization/src/main/java/org/wso2/carbon/andes/authorization/internal/AuthorizationServiceComponent.java (hunk 5 above)
Comment (sinthuja) [inline components/andes/org.wso2.carbon.andes.authorization/src/main/java/org/wso2/carbon/andes/authorization/internal/AuthorizationServiceComponent.java]: Fixed.

**Pull request merged**